### PR TITLE
fix(signals): export DeepSignal

### LINKED ISSUE #4372: [Bug] @ngrx/signals: DeepSignal is not exported
### Which @ngrx/* package(s) are the source of the bug?

signals

### Minimal reproduction of the bug/regression with instructions

I am using signalStore with a nested object and get the following error message from Webstorm and when compiling:

`The inferred type of 'MyService' cannot be named without a reference to 'node_modules/@ngrx/signals/src/deep-signal'. This is likely not portable. A type annotation is necessary.`

I can run it locally without any problems and everything works fine.

I use it in an angular library and not directly in a angular project.

If I set the `declaration` in `tsconfig.json` to `false`, it works.

If I add the entry export `{ DeepSignal } from './deep-signal'`; in `/node_modules/@ngrx/signals/src/index.d.ts`, the compilation also works.

### Expected behavior

No error message should appear when compiling (`ng build`).

Probably only `export { DeepSignal } from ‘./deep-signal’` has to be added to` index.ts`.

### Versions of NgRx, Angular, Node, affected browser(s) and operating system(s)

@ngrx/signals: 18.0.0-beta.1
Angular: 18.0.1

### Other information

_No response_

### I would be willing to submit a PR to fix this issue

- [X] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by timdeschryver**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #4372

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: modules/signals/src/index.ts [hunk 1]
```diff
@@ -10,3 +10,4 @@ export { withComputed } from './with-computed';
 export { withHooks } from './with-hooks';
 export { withMethods } from './with-methods';
 export { withState } from './with-state';
+export { DeepSignal } from './deep-signal';
```
Comment (markostanimirovic) [inline modules/signals/src/index.ts:13]: we could move this export to the top of the file to keep the alphabetical order

Comment (timdeschryver) [reply modules/signals/src/index.ts]: 👍 sounds good, thanks.

**Review by markostanimirovic — APPROVED** [review]

**Review by brandonroberts — APPROVED** [review]

**Comment by matheo** [comment]
@timdeschryver is it possible to export `toDeepSignal` too?
other open source libraries can take advantage of it without duplicating it :(

**Comment by rainerhahnekamp** [comment]
@matheo, is `signalState()` not already the `toDeepSignal` you are looking for?

**Comment by matheo** [comment]
@rainerhahnekamp thanks for the answer!
the requirement is to convert and keep track of a changing snapshot:
https://github.com/niklas-wortmann/xstate-angular/issues/13
so every time the state-machine is updated, we get a refreshed DeepSignal like here:
https://github.com/niklas-wortmann/xstate-angular/blob/main/lib/xstate-ngx/src/lib/useSelector.ts

it doesn't fit quite well right? we need a primitive processing there

**Comment by timdeschryver** [comment]
@matheo  will you be installing the `@ngrx/signals` package just to have the `toDeepSignal` method available?

I think it would be better to create a new issue with this request and explain the need why it will be useful to also export `toDeepSignal`. We try to keep our public API to a minimum to make it easier to change in the future without introducing breaking changes.

**Pull request merged**